lib/network_utils.ml

```diff
@@ -1163,3 +1200,196 @@ module Ethtool = struct
+			with _ -> false
+		in
+		try
+			Pervasiveext.finally (fun () ->
+				output_string oc (string_of_int num_vfs))
+				(fun () -> close_out oc);
+			if check_vf_numbers dev num_vfs then Ok Sysfs_successful
+			else begin
+					debug "%s: no exceptions but fail to set SRIOV" dev;
+					Error (Unknown "unable to enable via sysfs")
+			end
+		with
+		| Sys_error s when Xstringext.String.has_substr s "out of range of" -> Error Bus_out_of_range
+		| Sys_error s when Xstringext.String.has_substr s "not enough MMIO resources" -> Error Not_enough_mmio_resources
+		| e ->
+			debug "%s: set sriov error with exception %s" dev (Printexc.to_string e);
+			Error (Unknown "unable to enable via sysfs")
+
+	let get_driver dev =
+		match (Sysfs.get_driver_name dev) with
+		| Some driver -> driver
+		| None -> raise (Sys_error ("cannot get driver name for " ^ dev))
+
+	let write_conf_file driver content=
+		Unixext.write_string_to_file (Printf.sprintf "/etc/modprobe.d/%s.conf" driver) (String.concat "\n" content)
+
+	let enable_internal dev =
+		let config_t = get_lspci_output dev |> get_sriov_info in
+		let driver =  get_driver dev in
+		let has_probe_conf, need_rebuild_initrd, conf = parse_modprobe_conf driver config_t.max_vfs in
+		let enable_sriov_via_modprobe ()=
+			match has_probe_conf, need_rebuild_initrd with
+			| true, true ->
+				write_conf_file driver conf;
+				rebuild_initrd ()
+			| false, false -> 
+				let new_option_line = Printf.sprintf "options %s max_vfs=%s" driver (gen_options_for_maxvfs driver config_t.max_vfs) in
+				write_conf_file driver (conf @ [new_option_line]);
+				rebuild_initrd ()
+			| _ -> ()
+		in
+		if config_t.max_vfs = 0 then raise (Sys_error (Printf.sprintf "%s: do not have sriov capabilities" dev));
+		if config_t.num_vfs = 0 then begin
+			debug "enable SR-IOV on a device: %s that is disabled" dev;
+			match set_sriov_via_sysfs dev config_t.max_vfs with
+			| Ok _ as r -> r
+			| Error Unknown _ ->
+				debug "%s does not support sysfs interfaces, trying modprobe" dev;
+				enable_sriov_via_modprobe ();
+				Ok Modprobe_successful_requires_reboot
+			| Error _ as e -> e
+		end
+		else begin
+			debug "enable SR-IOV on a device: %s that has been already enabled" dev;
+			match has_probe_conf with
+			| false -> Ok Sysfs_successful
+			| true -> 
+				enable_sriov_via_modprobe ();
+				Ok Modprobe_successful
+		end
+
+	let disable_internal dev =
+		let best_effort_disable () = 
+			try
+				let driver = get_driver dev in
+				let has_probe_conf, need_rebuild_intrd, conf = parse_modprobe_conf driver 0 in
+				match has_probe_conf,need_rebuild_intrd with
+				| false, false -> set_sriov_via_sysfs dev 0; ()
+				| true, true ->
+					write_conf_file driver conf;
+					rebuild_initrd ()
+				| _ -> ()
+			with e -> debug "%s: best effort disable sriov with exn: %s" dev (Printexc.to_string e);()
+		in
+		best_effort_disable(); Ok Disable_successful
+
+	let make_vf_conf_internal pcibuspath (vf_info : Sriov.sriov_pci_t) =
```

Comment:
Any code manipulating types from the IDL, such as `Sriov.sriov_pci_t`, should go into network_server.ml. Functions in network_util.ml are meant to be simple wrappers of Linux tools, which are tied together in network_server.ml functions. Hence, this entire function should be part of `make_vf_config`.